### ISSUE TITLE
Data_oM: string/number comparers?

### ISSUE THREAD

**Issue by pawelbaran** [BHoM/BHoM#691]
In [Revit_oM](https://github.com/BHoM/Revit_Toolkit/tree/master/Revit_oM/Enums) there are enums that represent string and number comparison type (equal, greater, comparison type etc.). Shouldn't these be centralized, e.g. in Data_oM?

The next step could be centralized engine methods that would do the evaluation, including tolerance etc. In general, all that is already in place in Revit_Toolkit - a matter of porting across.

**Comment by rwemay** [comment]
Was talking about this earlier today, great one to get going - we want to use as input to the Linq methods + expose comparers for humans to input into adapters eventually, for full control.

**Comment by pawelbaran** [comment]
Could you please have a look at what has been implemented in the Revit_Toolkit and make sure that you happy with its current form? It would be great to get it moved in this milestone.

**Comment by rwemay** [comment]
I did have a quick look. I think rather than enums for string search options, I think I had a comparer class in mind, with inputs like ignoreCase, removeWhiteSpace, exactMatch, contains, doesNotContain inputs. Similar to the start made on Linq wrapping here https://github.com/BHoM/BHoM_Engine/pull/1312

**Comment by pawelbaran** [comment]
Yeah, makes sense. Current implementation in the Revit_Toolkit is rather primitive - in the context of your PR I believe it will be much better to resolve the general case centrally (with which you have already progressed nicely) and then to align the toolkit and leverage the engine.

**Comment by pawelbaran** [comment]
@IsakNaslundBh @FraserGreenroyd @alelom has that been centralised over time? I bumped against similar issue very recently, ended up writing my own comparer methods (again), which sounds a bit silly - my gut feeling is we should be able to build a universal solution. What do you think?

**Comment by IsakNaslundBh** [comment]
To the best of my knowledge, this has been no significant work on this centrally, no. Might be a good one to discuss and plan out to maybe centralise some of your work @pawelbaran .

Think it could be useful for the Library engine to add some more complex match methods to fetch items based on properties of the items.
